File: 2023/tests/day21.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day21');

const example1 = `
...........
.....###.#.
.###.##..#.
..#.#...#..
....#.#....
.##..S####.
.##..#...#.
.......##..
.##.#.####.
.##..##.##.
...........
`.trim();

const input = fs.readFileSync('inputs/input21.txt', 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example1, 6)).toBe(16);
  });

  test('solution', () => {
    expect(part1(input, 64)).toBe(3816);
  });
});

describe('part 2', () => {
  test('example 1', () => {
    expect(part2(example1, 6)).toBe(16);
  });

  test('example 2', () => {
    expect(part2(example1, 10)).toBe(50);
  });

  test('example 3', () => {
    expect(part2(example1, 50)).toBe(1594);
  });

  test('example 4', () => {
    expect(part2(example1, 100)).toBe(6536);
  });

  // test('example 5', () => {
  //   expect(part2(example1, 500)).toBe(167004);
  // });
  //
  // test('example 6', () => {
  //   expect(part2(example1, 1000)).toBe(668697);
  // });
  //
  // test('example 7', () => {
  //   expect(part2(example1, 5000)).toBe(16733044);
  // });

  test('solution', () => {
    expect(part2(input, 26501365)).toBe(634549784009844);
  });
});

File: 2019/src/test/java/aoc/Day21Test.java
package aoc;

import org.junit.jupiter.api.*;

import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day21Test {
    private static String input;
    private Day21 day;

    @BeforeAll
    public static void init() throws IOException {
        input = Files.readString(Paths.get("src/main/resources/day21.txt"), StandardCharsets.UTF_8);
    }

    @BeforeEach
    public void setup() {
        day = new Day21();
    }

    @Nested
    @DisplayName("Part 1")
    class Part1Tests {
        @Test
        void testSolution() {
            long result = day.part1(input);
            assertEquals(19357534L, result);
        }
    }

    @Nested
    @DisplayName("Part 2")
    class Part2Tests {
        @Test
        void testSolution() {
            long result = day.part2(input);
            assertEquals(1142814363L, result);
        }
    }
}